test/integration/extensions/controller/backupbucket_test.go

```diff
@@ -0,0 +1,298 @@
+		prepareAndRunTest(false)
+	})
+})
+
+func prepareAndRunTest(ignoreOperationAnnotation bool) {
+	By("setup and start manager")
+	mgr, err := createAndStartManager(ignoreOperationAnnotation)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("setup namespace for test")
+	namespace, err := generateNamespaceName()
+	Expect(err).NotTo(HaveOccurred())
+
+	runTest(mgr.GetClient(), namespace, ignoreOperationAnnotation)
+}
+
+func createAndStartManager(ignoreOperationAnnotation bool) (manager.Manager, error) {
+	mgrContext, mgrCancel := context.WithCancel(ctx)
+
+	framework.AddCleanupAction(func() {
+		defer func() {
+			By("stopping manager")
+			mgrCancel()
+		}()
+	})
+
+	mgr, err := manager.New(restConfig, manager.Options{})
+	if err != nil {
+		return nil, err
+	}
+
+	if err := extensionsv1alpha1.AddToScheme(mgr.GetScheme()); err != nil {
+		return nil, err
+	}
+
+	if err := backupbucket.AddToManagerWithOptions(mgr, ignoreOperationAnnotation); err != nil {
+		return nil, err
+	}
+
+	go func() {
+		if err := mgr.Start(mgrContext.Done()); err != nil {
+			panic(err)
+		}
+	}()
+
+	return mgr, nil
+}
+
+func generateNamespaceName() (string, error) {
+	suffix, err := utils.GenerateRandomStringFromCharset(5, "0123456789abcdefghijklmnopqrstuvwxyz")
+	if err != nil {
+		return "", err
+	}
+	return "gextlib-backupbucket-it--" + suffix, nil
+}
+
+func runTest(c client.Client, namespaceName string, ignoreOperationAnnotation bool) {
+	var (
+		namespace    *corev1.Namespace
+		cluster      *extensionsv1alpha1.Cluster
+		secret       *corev1.Secret
+		backupBucket *extensionsv1alpha1.BackupBucket
+	)
+
+	framework.AddCleanupAction(func() {
+		By("delete backupbucket")
+		Expect(client.IgnoreNotFound(c.Delete(ctx, backupBucket))).To(Succeed())
+
+		By("wait until backupbucket is deleted")
+		Expect(common.WaitUntilExtensionCRDeleted(
+			ctx,
+			c,
+			logger,
+			func() extensionsv1alpha1.Object { return &extensionsv1alpha1.BackupBucket{} },
+			extensionsv1alpha1.BackupBucketResource,
+			backupBucket.Namespace,
+			backupBucket.Name,
+			10*time.Second,
+			16*time.Minute,
+		)).NotTo(HaveOccurred())
+
+		Expect(c.Get(ctx, client.ObjectKey{Name: namespaceName}, namespace)).To(Succeed())
+		Expect(namespace.Annotations[fake.AnnotationKeyOperation]).To(Equal(fake.AnnotationValueOperationDelete))
+
+		By("delete secret, cluster and namespace")
+		Expect(client.IgnoreNotFound(c.Delete(ctx, secret))).To(Succeed())
+		Expect(client.IgnoreNotFound(c.Delete(ctx, cluster))).To(Succeed())
+		Expect(client.IgnoreNotFound(c.Delete(ctx, namespace))).To(Succeed())
+	})
+
+	By("create namespace for test execution")
+	namespace = &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespaceName}}
+	Expect(c.Create(ctx, namespace)).To(Succeed())
+
+	By("create cluster")
+	cluster = &extensionsv1alpha1.Cluster{ObjectMeta: metav1.ObjectMeta{Name: namespaceName}}
+	Expect(c.Create(ctx, cluster)).To(Succeed())
```

Comment:
When executing this test against my local Seed cluster this fails. The Cluster CRD is expected to have a Shoot, Seed and Cloudprofile set but this is not the case.
```
Cluster.extensions.gardener.cloud "gextlib-backupbucket-it--ajfyy" is invalid: [spec.cloudProfile: Invalid value: "null": spec.cloudProfile in body must be of type object: "null", spec.seed: Invalid value: "null": spec.seed in body must be of type object: "null", spec.shoot: Invalid value: "null": spec.shoot in body must be of type object: "null"]
```

Reply:
Strange, can you share the output of `kubectl get crd clusters.extensions.gardener.cloud -o yaml`?

Reply:
OK, I figured it out and will push an update, thanks!

Reply:
Thanks. What was the issue? I mean the test itself applies the CRD from the charts folder?

Reply:
No, this was missing (application of `Cluster` CRD - only the `BackupBucket` CRD was applied).